ballista/rust/executor/src/main.rs

```diff
@@ -112,6 +116,21 @@ async fn main() -> Result<()> {
         .context("Could not connect to scheduler")?;
 
     let scheduler_policy = opt.task_scheduling_policy;
+    let cleanup_ttl = opt.executor_cleanup_ttl;
```

Comment:
I wonder if it might be possible to use `NamedTempFile` or some other struct from `tempfile` (already a dependency): https://crates.io/crates/tempfile

There are at least two benefits:
1. The files are dropped immediately after they are no longer used so required intermediate diskspace is minimized
2. They can't be accidentally dropped while still in use (which perhaps affects long running queries)

I did something similar in DataFusion here: https://github.com/apache/arrow-datafusion/pull/1680

Reply:
@alamb Thanks for your advice 😊!
IMHP, if one job has 3 stage, stage2 read stage1 input then delete the file, but stage2 task fail,
In ballista, scheduler will start a task to reload stage1 input.  I think using  `NamedTempFile` will cause some trouble and complexity.  
we need keep the file for task-recovery and stage retry (like spark). So i decide if all the files under job_dir not modified in TTL we can safely delete it.
If i am not right,  Please correct me 🙈

Reply:
I don't honestly know enough about Ballista and its executor to know. 

What do you think @yahoNanJing  and @liukun4515 ?